src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotDeleteStorageAccountStep.java

```diff
@@ -0,0 +1,61 @@
+package bio.terra.service.snapshot.flight.delete;
+
+import bio.terra.service.dataset.flight.ingest.OptionalStep;
+import bio.terra.service.resourcemanagement.ResourceService;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import java.util.UUID;
+import java.util.function.Predicate;
+import org.elasticsearch.ResourceNotFoundException;
+
+public class DeleteSnapshotDeleteStorageAccountStep extends OptionalStep {
+
+  private UUID snapshotId;
+  private ResourceService resourceService;
+  private final AzureStorageAccountService storageAccountService;
+
+  public DeleteSnapshotDeleteStorageAccountStep(
+      UUID snapshotId,
+      ResourceService resourceService,
+      AzureStorageAccountService storageAccountService,
+      Predicate<FlightContext> doCondition) {
+    super(doCondition);
+    this.snapshotId = snapshotId;
+    this.resourceService = resourceService;
+    this.storageAccountService = storageAccountService;
+  }
+
+  @Override
+  public StepResult doOptionalStep(FlightContext context) throws InterruptedException {
+    FlightMap workingMap = context.getWorkingMap();
+    try {
+      AzureStorageAccountResource snapshotStorageAccountResource =
+          resourceService
+              .getSnapshotStorageAccount(snapshotId)
+              .orElseThrow(
+                  () -> new ResourceNotFoundException("Snapshot storage account not found"));
+
+      workingMap.put(
+          SnapshotWorkingMapKeys.STORAGE_ACCOUNT_RESOURCE_NAME,
+          snapshotStorageAccountResource.getName());
+
+      storageAccountService.deleteCloudStorageAccount(snapshotStorageAccountResource);
+
+    } catch (ResourceNotFoundException nfe) {
```

Comment:
I don't think it's a good practice to use an exception for local flow-of-control operations. If the only throw point for `ResourceNotFoundException` is line 41 above, it would be clearer to use an `if` and not `throw` and `catch`.

Reply:
True - let me know your thoughts on my changes

Reply:
Not sure I see what the change here was; I still see code that throws and then catches the same exception in a single block. If you want to keep the `Optional` functional style operation, you can use `ifPresent()`:
```
    resourceService
        .getSnapshotStorageAccount(snapshotId)
        .ifPresent(
            account -> {
              workingMap.put(
                  SnapshotWorkingMapKeys.STORAGE_ACCOUNT_RESOURCE_NAME, account.getName());
              storageAccountService.deleteCloudStorageAccount(account);
            });
```